host-bmc/dbus/pcie_topology.cpp

```diff
@@ -0,0 +1,144 @@
+
+            if (stateEffecterPDRs.empty())
+            {
+                std::cerr
+                    << "PCIe Topology: The state set PDR can not be found, entityType = "
+                    << entityType << std::endl;
+                return false;
+            }
+
+            uint16_t effecterID = 0;
+            for (auto& rep : stateEffecterPDRs)
+            {
+                auto pdr =
+                    reinterpret_cast<pldm_state_effecter_pdr*>(rep.data());
+                effecterID = pdr->effecter_id;
+                break;
+            }
+
+            hostEffecterParser->sendSetStateEffecterStates(
+                mctpEid, effecterID, 1, stateField,
+                std::bind(std::mem_fn(
+                              &pldm::dbus::PCIETopology::updateTopologyRefresh),
+                          this, std::placeholders::_1),
+                value);
+            return sdbusplus::com::ibm::PLDM::server::PCIeTopology::
+                pcIeTopologyRefresh(value);
+        }
+    }
+
+    return sdbusplus::com::ibm::PLDM::server::PCIeTopology::pcIeTopologyRefresh(
+        value);
+}
+
+bool PCIETopology::savePCIeTopologyInfo() const
+{
+    return sdbusplus::com::ibm::PLDM::server::PCIeTopology::
+        savePCIeTopologyInfo();
+}
+
+bool PCIETopology::savePCIeTopologyInfo(bool value)
+{
+    std::vector<set_effecter_state_field> stateField;
+
+    if (value ==
+        sdbusplus::com::ibm::PLDM::server::PCIeTopology::savePCIeTopologyInfo())
+    {
+        stateField.push_back({PLDM_NO_CHANGE, 0});
+        return sdbusplus::com::ibm::PLDM::server::PCIeTopology::
+            savePCIeTopologyInfo(value);
+    }
+    else
+    {
+        stateField.push_back({PLDM_REQUEST_SET, SAVE_PCIE_TOPLOGY});
+    }
+
+    if (value)
+    {
+        if (hostEffecterParser)
+        {
+            pldm::pdr::EntityType entityType = PLDM_ENTITY_GROUP | 0x8000;
+            auto stateEffecterPDRs = pldm::utils::findStateEffecterPDR(
+                mctpEid, entityType,
+                static_cast<uint16_t>(PLDM_OEM_IBM_PCIE_TOPOLOGY_ACTIONS),
+                hostEffecterParser->getPldmPDR());
+
+            if (stateEffecterPDRs.empty())
+            {
+                std::cerr
+                    << "PCIe Topology: The state set PDR can not be found, entityType = "
+                    << entityType << std::endl;
+                return false;
+            }
+
+            uint16_t effecterID = 0;
+            for (auto& rep : stateEffecterPDRs)
+            {
+                auto pdr =
+                    reinterpret_cast<pldm_state_effecter_pdr*>(rep.data());
+                effecterID = pdr->effecter_id;
+                break;
+            }
+
+            hostEffecterParser->sendSetStateEffecterStates(
+                mctpEid, effecterID, 1, stateField, nullptr, value);
+            return sdbusplus::com::ibm::PLDM::server::PCIeTopology::
+                savePCIeTopologyInfo(false);
```

Comment:
all of these seems like common code for both the properties , can we make a function for these ? and call it with parameters ?